docs/upgrade-guide.md

```diff
@@ -65,8 +66,8 @@ loggers for all AttributeManagers (i.e. for all layers).
 
 #### Removed method: `AttributeManager.addDynamic`
 
-This method has been deprecated since version 2.5 and is now removed, use
-`AttributeManager.add()` instead
+This method has been deprecated since version 2.5 and is now removed in v4.
+Use `AttributeManager.add()` instead.
```

Comment:
Good catch.

It was used in so many places and also when I started replacing it the app lines became a lot longer, forcing ugly line breaks. So... I let it be for the moment.

We should probably do a search for DEPRECATED in the code base and see if there are more old v3 things we can cut.